src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDocumentSynchronizationEndpoint.cs

```diff
@@ -78,84 +73,40 @@ public async Task<Unit> Handle(DidChangeTextDocumentParams notification, Cancell
             var sourceText = await document.GetTextAsync();
             sourceText = ApplyContentChanges(notification.ContentChanges, sourceText);
 
-            if (notification.TextDocument.Version is null)
-            {
-                throw new InvalidOperationException(RazorLS.Resources.Version_Should_Not_Be_Null);
-            }
-
             await _projectSnapshotManagerDispatcher.RunOnDispatcherThreadAsync(
-                () => _projectService.UpdateDocument(document.FilePath, sourceText, notification.TextDocument.Version.Value),
+                () => _projectService.UpdateDocument(document.FilePath, sourceText, notification.TextDocument.Version),
                 CancellationToken.None).ConfigureAwait(false);
 
             return Unit.Value;
         }
 
-        public async Task<Unit> Handle(DidOpenTextDocumentParams notification, CancellationToken token)
+        public async Task<Unit> Handle(DidOpenTextDocumentParamsBridge notification, CancellationToken token)
         {
             var sourceText = SourceText.From(notification.TextDocument.Text);
 
-            if (notification.TextDocument.Version is null)
-            {
-                throw new InvalidOperationException(RazorLS.Resources.Version_Should_Not_Be_Null);
-            }
-
             await _projectSnapshotManagerDispatcher.RunOnDispatcherThreadAsync(
-                () => _projectService.OpenDocument(notification.TextDocument.Uri.GetAbsoluteOrUNCPath(), sourceText, notification.TextDocument.Version.Value),
+                () => _projectService.OpenDocument(notification.TextDocument.Uri.GetAbsoluteOrUNCPath(), sourceText, notification.TextDocument.Version),
                 CancellationToken.None).ConfigureAwait(false);
 
             return Unit.Value;
         }
 
-        public async Task<Unit> Handle(DidCloseTextDocumentParams notification, CancellationToken token)
+        public async Task<Unit> Handle(DidCloseTextDocumentParamsBridge notification, CancellationToken token)
         {
             await _projectSnapshotManagerDispatcher.RunOnDispatcherThreadAsync(
                 () => _projectService.CloseDocument(notification.TextDocument.Uri.GetAbsoluteOrUNCPath()),
-                CancellationToken.None).ConfigureAwait(false);
+                token).ConfigureAwait(false);
```

Comment:
Is there any reason I shouldn't do this? I was operating under the assumption that it was just an oversight to not be this way before.

Reply:
Hmmm, good question. It's an interesting area because if the platform (or O#) cancels a didClose request we then have to ask ourselves, what's the current state of the world? Meaning, do we close the document do we not? If O# canceled it does that mean the servers shutting down or something else? If the platform canceled it are they now tracking the document as closed / should we? It opens up a suppppppper interesting can of worms that I'm not quite sure how we should act to be honest.

Now one bit I do like about passing the token is that if we get a report from a user that funky stuff be happening and this now throws (because we're passing the token) it'll be present in the logs